application/org.openjdk.jmc.console.twitter/src/main/java/org/openjdk/jmc/console/twitter/TwitterPlugin.java

```diff
@@ -344,4 +367,44 @@ public boolean verifyAuthorizedUser(String username) {
 			return false;
 		}
 	}
+
+	private void verifyTweeter(String fromUser) throws Exception {
+		Tweeter tweeter = getAuthorizedTweeter(fromUser);
+		if (tweeter != null) {
+			AccessToken at = createAccessToken(tweeter);
+			oAuthHeaderGenerator = new TwitterOAuthHeaderGenerator(consumerKey, consumerSecret, at.getAccessToken(),
+					at.getAccessTokenSecret());
+		} else {
+			throw new Exception(String.format(
+					"Attempted to send direct message from account not defined in the preferences. Please set up the account (%s) in preferences first.",
+					fromUser));
+		}
+	}
+
+	public void sendDirectMessage(String from, String to, String message) throws Exception {
+		verifyTweeter(from);
+		Map<String, String> requestParams = new HashMap<String, String>();
+		String json = createJSONObject(getUserId(to), message);
+		HttpRequest request = HttpRequest.newBuilder().POST(HttpRequest.BodyPublishers.ofString(json))
+				.uri(URI.create(SEND_DIRECT_MESSAGE_URL))
+				.setHeader("Authorization", getHeader("POST", SEND_DIRECT_MESSAGE_URL, requestParams))
+				.header("Content-Type", "application/json").build();
+
+		HttpResponse<String> response = getHttpClient().send(request, HttpResponse.BodyHandlers.ofString());
+		if (response.statusCode() != 200) {
+			JOptionPane.showMessageDialog(null,
+					" Some error occured while sending direct message. Please verify your twitter app settings. ");
```

Comment:
There are trailing spaces at the beginning and end of this string. It would also be nice to have the error status included with this string, so the user has a better idea of where an issue might have happened if it occurs. The string in the dialog might be something like:

"An error occured while __________. Please verify your Twitter app settings. (Error code: XXX)"